spec/spec_helper_local.rb

```diff
@@ -1,6 +1,33 @@
-require 'rspec-puppet-facts'
-include RspecPuppetFacts
-#                                                                       # Original fact sources:
-add_custom_fact :puppetversion, Puppet.version                          # Facter, but excluded from rspec-puppet-facts
 add_custom_fact :rabbitmq_version, '3.6.1'                              # puppet-rabbitmq
 add_custom_fact :erl_ssl_path, '/usr/lib64/erlang/lib/ssl-7.3.3.1/ebin' # puppet-rabbitmq
+
+def os_specific_facts(facts)
```

Comment:
Alternative:
```ruby
add_custom_fact :systemd, lambda do |os, facts|
  case facts[:os]['family']
  when 'Debian'
    case facts[:os]['name']
    when 'Debian'
      facts[:os]['release']['major'].to_i >= 8
    when 'Ubuntu'
      facts[:os]['release']['major'].to_i >= 16
    else
      true
    end
  when 'RedHat'
    facts[:os]['release']['major'].to_i >= 7
  else
    true
  end
end
```

The benefit is that you don't have to merge it into `facts` in every test but it's just present. You do need to duplicate it for `service_provider` if you want that to be present. We don't use that now AFAIK so I'd leave it out. If it's important, you could probably rewrite it to use the `service_provider` fact:
```ruby
add_custom_fact :systemd, lambda { |os, facts| facts[:service_provider] == 'systemd' }
```

Reply:
service_provider fact comes from stdlib and isn't available in facterdb :( I will update it to your first approach and also propose it to modulesync_config

Reply:
for now I would like to stick with the current version and implement something nice for both facts in modulsync_config. I am not sure where os and facts should come from in your code and it give me an error:

```
ArgumentError:
  tried to create Proc object without a block
```